python/src/scipp/compat/mantid.py

```diff
@@ -111,44 +111,53 @@ def set_bin_masks(bin_masks, dim, index, masked_bins):
         bin_masks[sc.Dim.Spectrum, index][dim, masked_bin].value = True
 
 
+def validate_and_get_unit(unit):
+    known_units = {
+        "DeltaE": [sc.Dim.EnergyTransfer, sc.units.meV],
+        "TOF": [sc.Dim.Tof, sc.units.us],
+        "Wavelength": [sc.Dim.Wavelength, sc.units.angstrom],
+        "Energy": [sc.Dim.Energy, sc.units.meV],
+        "dSpacing": [sc.Dim.DSpacing, sc.units.angstrom],
+        "MomentumTransfer": [sc.Dim.Q, sc.units.dimensionless / sc.units.angstrom],
+        "QSquared": [sc.Dim.Q, sc.units.dimensionless / (sc.units.angstrom * sc.units.angstrom)],
+        "Label": [sc.Dim.Spectrum, sc.units.dimensionless],
+    }
+
+    if unit not in known_units.keys():
+        raise RuntimeError("X-axis unit not currently supported."
+                           "Possible values are: {}, "
+                           "got '{}'. ".format(
+                               [k for k in known_units.keys()], unit))
+    else:
+        return known_units[unit]
+
+
 def convert_Workspace2D_to_dataarray(ws):
     common_bins = ws.isCommonBins()
     comp_info = make_component_info(ws)
     det_info = make_detector_info(ws)
     pos, num = init_pos_spectrum_no(ws)
 
-    # TODO More cases?
-    allowed_units = {
-        "DeltaE": [sc.Dim.EnergyTransfer, sc.units.meV],
-        "TOF": [sc.Dim.Tof, sc.units.us],
-        "Wavelength": [sc.Dim.Wavelength, sc.units.angstrom]
-    }
-    xunit = ws.getAxis(0).getUnit().unitID()
-    if xunit not in allowed_units.keys():
-        raise RuntimeError("X-axis unit not currently supported for "
-                           "Workspace2D. Possible values are: {}, "
-                           "got '{}'. ".format(
-                               [k for k in allowed_units.keys()], xunit))
-    else:
-        [dim, unit] = allowed_units[xunit]
+    dim, unit = validate_and_get_unit(ws.getAxis(0).getUnit().unitID())
+    spec_dim, _ = validate_and_get_unit(ws.getAxis(1).getUnit().unitID())
```

Comment:
Why no unit for this axis?

Edit: Hmm, I see, because we are still using just spectrum numbers for this coord. Is it possible to do something smarter, i.e., extract more information about axis 1?